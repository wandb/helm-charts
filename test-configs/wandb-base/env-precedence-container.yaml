global:
  extraEnv:
    EXAMPLE: "100"
  env:
    EXAMPLE: "200"

extraEnv:
  EXAMPLE: "300"

env:
  EXAMPLE: "400"

size: "small"

sizing:
  small:
    env:
      EXAMPLE: "500"

kind: Job
jobs:
  test:
    containers:
      example:
        image:
          repository: debian
          tag: 12-slim
        command:
          - /bin/bash
        args:
          - -c
          - |
            echo "EXAMPLE=$EXAMPLE"
            EXPECTED="600"
            if [ "$EXAMPLE" = "$EXPECTED" ]; then
              echo "Value is $EXPECTED"
              exit 0
            else
              echo "Value is NOT $EXPECTED"
              exit 1
            fi
        env:
          EXAMPLE: "600"
