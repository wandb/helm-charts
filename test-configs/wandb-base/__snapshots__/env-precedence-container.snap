# chartsnap: snapshot_version=v3
---
# Source: wandb-base/templates/pdb.yaml
apiVersion: policy/v1
kind: PodDisruptionBudget
metadata:
  name: chartsnap-wandb-base
  labels:
    helm.sh/chart: '###CHART_VERSION###'
    app.kubernetes.io/name: wandb-base
    app.kubernetes.io/instance: chartsnap
    app.kubernetes.io/managed-by: Helm
spec:
  selector:
    matchExpressions:
    - key: batch.kubernetes.io/job-name
      operator: DoesNotExist
    matchLabels:
      app.kubernetes.io/name: wandb-base
      app.kubernetes.io/instance: chartsnap
  maxUnavailable: 1
---
# Source: wandb-base/templates/serviceaccount.yaml
apiVersion: v1
kind: ServiceAccount
metadata:
  name: chartsnap-wandb-base
  labels:
    helm.sh/chart: '###CHART_VERSION###'
    app.kubernetes.io/name: wandb-base
    app.kubernetes.io/instance: chartsnap
    app.kubernetes.io/managed-by: Helm
automountServiceAccountToken: true
---
# Source: wandb-base/templates/job.yaml
apiVersion: batch/v1
kind: Job
metadata:
  name: chartsnap-test
  labels:
    helm.sh/chart: '###CHART_VERSION###'
    app.kubernetes.io/name: wandb-base
    app.kubernetes.io/instance: chartsnap
    app.kubernetes.io/managed-by: Helm
spec:
  template:
    metadata:
      labels:
        helm.sh/chart: wandb-base-0.11.0
        app.kubernetes.io/name: wandb-base
        app.kubernetes.io/instance: chartsnap
        app.kubernetes.io/managed-by: Helm
    spec:
      containers:
      - name: example
        command:
        - /bin/bash
        args:
        - -c
        - |
          echo "EXAMPLE=$EXAMPLE"
          EXPECTED="600"
          if [ "$EXAMPLE" = "$EXPECTED" ]; then
            echo "Value is $EXPECTED"
            exit 0
          else
            echo "Value is NOT $EXPECTED"
            exit 1
          fi
        envFrom:
        env:
        - name: EXAMPLE
          value: "600"
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            add: []
            drop: []
          privileged: false
          readOnlyRootFilesystem: false
        image: "debian:12-slim"
        imagePullPolicy:
      restartPolicy: Never
      serviceAccountName: chartsnap-wandb-base
      securityContext:
        fsGroup: 0
        fsGroupChangePolicy: OnRootMismatch
        runAsGroup: 0
        runAsNonRoot: true
        runAsUser: 999
        seccompProfile:
          type: RuntimeDefault
      topologySpreadConstraints:
      - labelSelector:
          matchLabels:
            app.kubernetes.io/instance: chartsnap
            app.kubernetes.io/name: wandb-base
        maxSkew: 1
        topologyKey: topology.kubernetes.io/zone
        whenUnsatisfiable: ScheduleAnyway
      - labelSelector:
          matchLabels:
            app.kubernetes.io/instance: chartsnap
            app.kubernetes.io/name: wandb-base
        maxSkew: 1
        topologyKey: kubernetes.io/hostname
        whenUnsatisfiable: ScheduleAnyway
